lib/charms/postgresql_k8s/v0/postgresql.py

```diff
@@ -347,15 +347,21 @@ def set_up_database(self) -> None:
         """Set up postgres database with the right permissions."""
         connection = None
         try:
+            self.create_user(
+                "admin",
+                extra_user_roles="pg_read_all_data,pg_read_all_settings,pg_read_all_stats,pg_stat_scan_tables,pg_monitor,pg_signal_backend",
```

Comment:
I would limit privs by read and write all databases.
```suggestion
                extra_user_roles="pg_read_all_data,pg_write_all_data",
```

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/201/commits/3b9591933d20ef01dddf3429181ac2a2f9607e6b.